src/supernova/test.rs

```diff
@@ -80,40 +82,83 @@ where
     }
   }
 }
+/// c = a + b where a, b is AllocatedNum
+pub fn add_allocated_num<F: PrimeField, CS: ConstraintSystem<F>>(
+  mut cs: CS,
+  a: &AllocatedNum<F>,
+  b: &AllocatedNum<F>,
+) -> Result<AllocatedNum<F>, SynthesisError> {
+  let c = AllocatedNum::alloc(cs.namespace(|| "c"), || {
+    Ok(*a.get_value().get()? + b.get_value().get()?)
+  })?;
+  cs.enforce(
+    || "Check u_fold",
+    |lc| lc + a.get_variable() + b.get_variable(),
+    |lc| lc + CS::one(),
+    |lc| lc + c.get_variable(),
+  );
+  Ok(c)
+}
+
+fn next_rom_index_and_pc<F: PrimeField, CS: ConstraintSystem<F>>(
```

Comment:
This would  benefit from a doctoring no?

Reply:
I'm not sure what you mean. The name? I'm not keen on getting too deep into design/engineering of the 'API' for the toy ROM example, since it's pretty contrived in its current state anyway. But I may also miss what you are suggesting.